google/resource_storage_bucket.go

```diff
@@ -77,6 +79,65 @@ func resourceStorageBucket() *schema.Resource {
 				ForceNew: true,
 			},
 
+			"lifecycle_rule": {
+				Type:     schema.TypeList,
+				Optional: true,
+				Elem: &schema.Resource{
+					Schema: map[string]*schema.Schema{
+						"action": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Set:      resourceGCSBucketLifecycleRuleActionHash,
+							Elem: &schema.Resource{
+								Schema: map[string]*schema.Schema{
+									"type": {
+										Type:     schema.TypeString,
+										Required: true,
+									},
+									"storage_class": {
+										Type:     schema.TypeString,
+										Optional: true,
+									},
+								},
+							},
+						},
+						"condition": {
+							Type:     schema.TypeSet,
+							Required: true,
+							MaxItems: 1,
+							Set:      resourceGCSBucketLifecycleRuleConditionHash,
```

Comment:
Similarly, in theory I don't think this is required, but I don't see much harm in including it.